### PR TITLE
Adds mixed secret to the config

### PR DESCRIPTION
**Pull request by printer16**
Turns mixed secret into a config option and fixes a bug of it showing up in gamemode probabilities when it's off.

### CODE REVIEW

**Comment by printer16** [comment]
!review

File: code/controllers/configuration.dm [hunk 1]
```diff
@@ -505,6 +505,10 @@ var/list/gamemode_cache = list()
 				if ("server")
 					config.server = value
 
+				if("mixed_secret_enabled")
+					config.mixedsecret_enabled = TRUE
```
Comment (NonQueueingMatt) [inline code/controllers/configuration.dm:509]: should probably change this to mixed_secret_enabled as well

File: code/controllers/configuration.dm [hunk 2]
```diff
@@ -53,6 +53,7 @@ var/list/gamemode_cache = list()
 	var/list/votable_modes = list()		// votable modes
 	var/list/probabilities_secret = list()			// relative probability of each mode in secret/random
 	var/list/probabilities_mixed_secret = list()	// relative probability of each mode in heavy secret mode
+	var/mixedsecret_enabled = FALSE
```
Comment (Geevies) [inline code/controllers/configuration.dm:56]:
```suggestion
	var/mixed_secret_enabled = FALSE
```

File: code/controllers/configuration.dm (hunk 1 above)
Comment (Geevies) [inline code/controllers/configuration.dm:509]:
```suggestion
					config.mixed_secret_enabled = TRUE
```

File: code/controllers/subsystems/vote.dm [hunk 3]
```diff
@@ -245,7 +245,7 @@ var/datum/controller/subsystem/vote/SSvote
 						continue
 					AddChoice(F, capitalize(M.name), "[M.required_players]")
 				AddChoice(ROUNDTYPE_STR_SECRET, "Secret")
-				if(ROUNDTYPE_STR_MIXED_SECRET in choices)
+				if(config.mixedsecret_enabled && (ROUNDTYPE_STR_MIXED_SECRET in choices))
```
Comment (Geevies) [inline code/controllers/subsystems/vote.dm:248]:
```suggestion
				if(config.mixed_secret_enabled && (ROUNDTYPE_STR_MIXED_SECRET in choices))
```

File: code/game/gamemodes/game_mode.dm [hunk 4]
```diff
@@ -770,13 +770,14 @@ proc/get_nt_opposed()
...
 
-		to_chat(src, "<b>Mixed Secret Mode Odds:</b>")
-		sum = 0
-		for(var/config_tag in config.probabilities_mixed_secret)
-			sum += config.probabilities_mixed_secret[config_tag]
-		for(var/config_tag in config.probabilities_mixed_secret)
-			if(config.probabilities_mixed_secret[config_tag] > 0)
-				var/percentage = round(config.probabilities_mixed_secret[config_tag] / sum * 100, 0.1)
-				to_chat(src, "[config_tag] [percentage]%")
+		if(config.mixedsecret_enabled)
```
Comment (Geevies) [inline code/game/gamemodes/game_mode.dm:773]:
```suggestion
		if(config.mixed_secret_enabled)
```

**Comment by skull132** [comment]
Unaddressed changes for a long time. RIPeroning.

Reopen once you've fixed these.

**Pull request closed without merging**